### PR TITLE
Add azure_dns_zone table closes #564

### LINKED ISSUE #564: Add table azure_dns_zone
**References**
- Terraform resource: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/dns_zone
- Azure SDK Go: https://github.com/Azure/azure-sdk-for-go/tree/v68.0.0/services/dns/mgmt/2018-05-01/dns
- Models/attributes: https://github.com/Azure/azure-sdk-for-go/blob/v68.0.0/services/dns/mgmt/2018-05-01/dns/models.go#L597

### PR DESCRIPTION
**Pull request by pdecat**
Resolves #564 

TODO:
- [x] handle pagination (limited to 100 zones right now)
- [x] add documentation 
- [x] run extensive tests
- [x] add integration tests

### CODE REVIEW

File: azure/table_azure_dns_zone.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+	if err != nil {
+		return nil, err
+	}
+	subscriptionID := session.SubscriptionID
+
+	dnsClient := dns.NewZonesClientWithBaseURI(session.ResourceManagerEndpoint, subscriptionID)
+	dnsClient.Authorizer = session.Authorizer
+
+	// ListAll API doesn't exist
+	result, err := dnsClient.List(ctx, nil) // FIXME: handle pagination as this returns only 100 zones
```
Comment (pdecat) [inline azure/table_azure_dns_zone.go:144]: FIXME: handle pagination as this returns only 100 zones

Comment (pdecat) [reply azure/table_azure_dns_zone.go]: Pagination is actually already handled below.

File: azure/table_azure_dns_zone.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
+)
+
+//// TABLE DEFINITION
+
+func tableAzureDNSZone(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "azure_dns_zone",
+		Description: "Azure DNS zone",
```
Comment (misraved) [inline azure/table_azure_dns_zone.go:18]:
```suggestion
		Description: "Azure DNS Zone",
```

File: azure-test/tests/azure_dns_zone/variables.tf [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+resource "azurerm_resource_group" "named_test_resource" {
+  name     = var.resource_name
+  location = "East US"
+  tags = {
+    name = var.resource_name
+  }
+}
+
+resource "azurerm_dns_zone" "named_test_resource" {
+  name                = "${var.resource_name}.xyz"
```
Comment (misraved) [inline azure-test/tests/azure_dns_zone/variables.tf:44]:
```suggestion
  name                = "${var.resource_name}"
```
Do we need `.xyz` references?

**Review by misraved — COMMENTED** [review]
Thanks @pdecat for creating this new table. I have left minor review comments, please let me know if you have questions.

Also, please rebase this branch with the main, since we recently updated all the tables to use the Steampipe plugin SDK v5.

PR reference - https://github.com/turbot/steampipe-plugin-azure/pull/574

File: azure-test/tests/azure_dns_zone/variables.tf (hunk 3 above)
Comment (pdecat) [reply azure-test/tests/azure_dns_zone/variables.tf:44]: Yes, it is required to be a valid DNS name with at least two parts.

File: azure/table_azure_dns_zone.go [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+			},
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listDNSZones,
+		},
+		Columns: azureColumns([]*plugin.Column{
+			{
+				Name:        "name",
+				Type:        proto.ColumnType_STRING,
+				Description: "The friendly name that identifies the DNS zone",
```
[the remark this answers is not in the document]
Comment (pdecat) [reply azure/table_azure_dns_zone.go]: Done.

**Comment by pdecat** [comment]
Hi @misraved, I believe I've addressed all your comments. Could you give it another look? :pray:

**Comment by pdecat** [comment]
New integration logs:

```
./tint.js azure_dns_zone
No env file present for the current environment:  staging
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/azure_dns_zone []

PRETEST: tests/azure_dns_zone

TEST: tests/azure_dns_zone
Running terraform
data.azurerm_client_config.current: Reading...
data.azurerm_client_config.current: Read complete after 0s [id=Y2xpZW50Q29uZmlncy9jbGllbnRJZD0wNGIwNzc5NS04ZGRiLTQ2MWEtYmJlZS0wMmY5ZTFiZjdiNDY7b2JqZWN0SWQ9MTc2YjQ0Y2MtNDEzZi00MmFiLTk5MzgtMzYwYzU1YjgyZTk2O3N1YnNjcmlwdGlvbklkPWFlODk2NzNkLWYzNTMtNDQ4ZC1iMzY5LTllYWNiMGNlNzc0Mzt0ZW5hbnRJZD1mODdhNzY0MC0zYjk0LTRiYmQtYjVmYS1iNGMxNTk0N2NmNTY=]
data.null_data_source.resource: Reading...
data.null_data_source.resource: Read complete after 0s [id=static]

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # azurerm_dns_zone.named_test_resource will be created
  + resource "azurerm_dns_zone" "named_test_resource" {
      + id                        = (known after apply)
      + max_number_of_record_sets = (known after apply)
      + name                      = "turbottest22547.xyz"
      + name_servers              = (known after apply)
      + number_of_record_sets     = (known after apply)
      + resource_group_name       = "turbottest22547"
      + tags                      = {
          + "name" = "turbottest22547.xyz"
        }

      + soa_record {
          + email         = (known after apply)
          + expire_time   = (known after apply)
          + fqdn          = (known after apply)
          + host_name     = (known after apply)
          + minimum_ttl   = (known after apply)
          + refresh_time  = (known after apply)
          + retry_time    = (known after apply)
          + serial_number = (known after apply)
          + tags          = (known after apply)
          + ttl           = (known after apply)
        }
    }

  # azurerm_resource_group.named_test_resource will be created
  + resource "azurerm_resource_group" "named_test_resource" {
      + id       = (known after apply)
      + location = "eastus"
      + name     = "turbottest22547"
      + tags     = {
          + "name" = "turbottest22547"
        }
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + resource_aka    = (known after apply)
  + resource_id     = (known after apply)
  + resource_name   = "turbottest22547"
  + subscription_id = "******"
azurerm_resource_group.named_test_resource: Creating...
azurerm_resource_group.named_test_resource: Creation complete after 2s [id=/subscriptions/******/resourceGroups/turbottest22547]
azurerm_dns_zone.named_test_resource: Creating...
azurerm_dns_zone.named_test_resource: Creation complete after 8s [id=/subscriptions/******/resourceGroups/turbottest22547/providers/Microsoft.Network/dnsZones/turbottest22547.xyz]

Warning: Deprecated

  with data.null_data_source.resource,
  on variables.tf line 29, in data "null_data_source" "resource":
  29: data "null_data_source" "resource" {

The null_data_source was historically used to construct intermediate values
to re-use elsewhere in configuration, the same can now be achieved using
locals

(and one more similar warning elsewhere)

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

resource_aka = "azure:///subscriptions/******/resourceGroups/turbottest22547/providers/Microsoft.Network/dnszones/turbottest22547.xyz"
resource_id = "/subscriptions/******/resourceGroups/turbottest22547/providers/Microsoft.Network/dnszones/turbottest22547.xyz"
resource_name = "turbottest22547"
subscription_id = "******"

Running SQL query: test-get-query.sql
[
  {
    "id": "/subscriptions/******/resourceGroups/turbottest22547/providers/Microsoft.Network/dnszones/turbottest22547.xyz",
    "name": "turbottest22547.xyz",
    "region": "global",
    "type": "Microsoft.Network/dnszones"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "akas": [
      "azure:///subscriptions/******/resourceGroups/turbottest22547/providers/Microsoft.Network/dnszones/turbottest22547.xyz",
      "azure:///subscriptions/******/resourcegroups/turbottest22547/providers/microsoft.network/dnszones/turbottest22547.xyz"
    ],
    "name": "turbottest22547.xyz",
    "tags": {
      "name": "turbottest22547.xyz"
    },
    "title": "turbottest22547.xyz"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "id": "/subscriptions/******/resourceGroups/turbottest22547/providers/Microsoft.Network/dnszones/turbottest22547.xyz",
    "name": "turbottest22547.xyz"
  }
]
✔ PASSED

POSTTEST: tests/azure_dns_zone

TEARDOWN: tests/azure_dns_zone

SUMMARY:

1/1 passed.
```

File: azure-test/tests/azure_dns_zone/variables.tf (hunk 3 above)
Comment (pdecat) [reply azure-test/tests/azure_dns_zone/variables.tf:44]: FWIW, I took the `.xyz` from Azure documentation examples: https://learn.microsoft.com/en-us/azure/dns/dns-getstarted-portal

**Review by Shr3ps — APPROVED** [review]

**Review by misraved — APPROVED** [review]

**Pull request merged**